app/src/main/java/org/mozilla/focus/utils/DialogUtils.java

```diff
@@ -174,9 +175,16 @@ public static void showRateAppNotification(Context context) {
     public static void showDefaultSettingNotification(Context context) {
 
         // Let NotificationActionBroadcastReceiver handle what to do
-        final Intent openDefaultBrowserSetting = IntentUtils.genDefaultBrowserSettingIntentForBroadcastReceiver(context);
-        final PendingIntent openRocketPending = PendingIntent.getBroadcast(context, REQUEST_DEFAULT_CLICK, openDefaultBrowserSetting,
-                PendingIntent.FLAG_ONE_SHOT);
+        final Intent openDefaultBrowserSetting;
+        final PendingIntent openRocketPending;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            openDefaultBrowserSetting = new Intent(android.provider.Settings.ACTION_MANAGE_DEFAULT_APPS_SETTINGS);
+            openRocketPending = PendingIntent.getActivity(context, REQUEST_DEFAULT_CLICK, openDefaultBrowserSetting, PendingIntent.FLAG_ONE_SHOT);
```

Comment:
If we open the setting directly, can we still monitor/send Telemetry to track the click action of that notification if API level >= 24?
see:
https://github.com/mozilla-tw/Rocket/blob/567ebbe3b04dea34bc28200e4845da9e8354c952/app/src/main/java/org/mozilla/focus/notification/NotificationActionBroadcastReceiver.java#L73

Reply:
Yes, you are right. Good catch, the current implementation is as expected.